blog/2025-07-01-tigrisfs/index.mdx

```diff
@@ -0,0 +1,368 @@
+shells with the new world of dynamic infinity in the cloud.
+
+![](./linux-diagram.webp)
+
+In general, tigrisfs can be slower than the native disk for files that aren’t
+cached yet, but it more than makes up for it by allowing you to make the
+location of your files irrelevant. All you need to do is run tigrisfs, and you
+have a single global namespace for your data across all your machines.
+
+tigrisfs is written in Go and is
+[open source on GitHub](https://github.com/tigrisdata/tigrisfs). We welcome any
+and all contributions to make it even better\!
+
+For most file operations, tigrisfs will be more than enough. If you need special
+Unix features like Unix sockets, character devices, or other magic things like
+extended attributes, tigrisfs won’t work out for you.
+
+### Benchmarks
+
+Benchmarking filesystems is kind of annoying, and networked filesystems can be
+even more annoying to benchmark. Most of the time, you end up making a lot of
+assumptions about the system state and network configuration. Here are the specs
+of our benchmarking machine:
+
+| Component         | Quantity                           |
+| ----------------- | ---------------------------------- |
+| Instance type     | VM.Standard.E5.Flex (Oracle Cloud) |
+| CPU cores         | 24                                 |
+| Memory            | 24 gigabytes (24Gi)                |
+| Network bandwidth | 24 gigabits per second             |
+
+Our benchmarks are done with the
+[flexible i/o tester fio](https://github.com/axboe/fio). Note that we are using
+direct I/O to avoid [page caching](https://en.wikipedia.org/wiki/Page_cache)
+being an issue.
+
+#### Read performance
+
+Here is the command we used to test read performance on a bucket:
+
+```sh
+fio --name=read_throughput \
+    --directory=/mnt/test-tigrisfs-bucket \
+    --numjobs=4 \
+    --size=4G \
+    --time_based \
+    --runtime=120s \
+    --ramp_time=2s \
+    --ioengine=libaio \
+    --direct=1 \
+    --verify=0 \
+    --bs=1M \
+    --iodepth=1 \
+    --rw=read \
+    --group_reporting=1
+```
+
+This has fio run for 2 minutes with 2 seconds of ramp-up time (during which, the
+results are not counted in the statistics) trying to read up to 4 gigabytes of
+data per thread (job) in one megabyte blocks. This reads a total of 16 gigabytes
+of data. The test was run in permutations of thread count and block size to see
+if the limitations are on tigrisfs, the Tigris service, and the network card of
+the machine.
+
+And we got these results for each permutation of the test:
+
+| Threads | Block Size | Throughput (MiB/sec) |
+| ------- | ---------- | -------------------- |
+| 4       | 1M         | 1630                 |
+| 4       | 4M         | 2446                 |
+| 8       | 1M         | 2802 \*              |
+| 8       | 4M         | 2732 \*              |
+
+The throughput numbers with the asterisk next to them could theoretically be
```

Comment:
Let's wrap it in a note block to make it more prominent

Reply:
Done in https://github.com/tigrisdata/tigris-blog/pull/229/commits/7a7ff010485965410781e75daf910206435013b7